### PR TITLE
Update logging to 0.2.3 for Windows 1909 support

### PR DESCRIPTION
**Pull request by nickgerace**
Update logging to 0.2.3 for Windows 1909 support. Includes other fixes as well.

Relevant issue: https://github.com/rancher/rancher/issues/30279
Relevant QA issue: https://github.com/rancher/rancher/issues/30568
Backport PR: https://github.com/rancher/system-charts/pull/390

~~**Note**: the Windows logging pod(s) can take up to 4-5 minutes to go into a `Running` state.~~
EDIT: This has been fixed with `DirectoryOrCreate`. Thanks to @brianharwell for the idea!

### CODE REVIEW

**Review by rmweir — DISMISSED** [review]
LGTM

**Comment by nickgerace** [comment]
Validated this chart with Windows 1909 nodes and ElasticSearch.

**Review by aiyengar2 — DISMISSED** [review]
LGTM

File: charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ spec:
     spec:
       serviceAccountName: {{ template "fluentd.fullname" . }}
       initContainers:
-      - name: init-{{ template "fluentd.fullname" . }}
+      - name: init-{{ template "fluentd.fullname" . }}-01
```
Comment (cbron) [inline charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Whats the actual purpose of these 2 init containers ? `01` and `02` don't explain.

File: charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml [hunk 2]
```diff
@@ -45,6 +45,15 @@ spec:
...
           name: rancher
+      - name: init-{{ template "fluentd.fullname" . }}-02
+        image:  {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
+        command:
+          - 'powershell'
+          - '-command'
+          - 'New-Item -ItemType Directory -Force -Path /var/lib/rancher/rke/log'
+        volumeMounts:
+        - mountPath: /var/lib/rancher
+          name: rancher
```
Comment (cbron) [inline charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Want to confirm that we want to mount same volume in both containers.

Comment (nickgerace) [reply charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Yes, we do

File: charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml (hunk 1 above)
Comment (nickgerace) [reply charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Two separate powershell commands need to be ran, so we can either write a PS1 script to run them in one container, or have an init container for each command.

The purpose of each one is to create a different directory path (akin to `mkdir -p path/to/desired/directory`).

Comment (nickgerace) [reply charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: I'll rename them to be clear on what they do.

Comment (nickgerace) [reply charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Done

Comment (cbron) [reply charts/rancher-logging/0.2.3/charts/fluentd/charts/fluentd-windows/templates/daemonset.yaml]: Much better

**Comment by nickgerace** [comment]
Confirmed that this chart works with my ES cluster for two version.

- 1909 SAC
- 2019 LTS (based on 1809, but not the same as 1809 SAC)

_(Side note: 1809 SAC and 1903 SAC are out of [MSFT support](https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info))_

**Review by cbron — APPROVED** [review]

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Comment by aiyengar2** [comment]
@nickgerace will this be backported to 2.4 as well?

**Comment by nickgerace** [comment]
Yes. @aiyengar2

**Pull request merged**